src/client.rs

```diff
@@ -374,7 +374,8 @@ impl Client {
         self.auth_store.read().await.contains_key(registry)
     }
 
-    async fn store_auth_if_needed(&self, registry: &str, auth: &RegistryAuth) {
+    /// Store the authentication information for this registry if it's not already stored in the client
```

Comment:
Can you add a bit more about when you'd use this? You could probably copy/paste most of it from the PR description

Reply:
Sure, I added a bit more info about when you need this. :+1: